tests/test_pydss_examples.py

```diff
@@ -0,0 +1,150 @@
+
+import os
+import sys
+import shutil
+import tempfile
+from distutils.dir_util import copy_tree
+
+import pytest
+import subprocess
+from PyDSS.pydss_project import PyDssProject
+
+
+
+PATH = os.path.join(tempfile.gettempdir(), "pydss-projects")
+EXAMPLES_path = "examples"
+
+@pytest.fixture
+def pydss_project():
+    if os.path.exists(PATH):
+        shutil.rmtree(PATH)
+    yield
+    if os.path.exists(PATH):
+        shutil.rmtree(PATH)
+
+
+base_projects_path = None
+def copy_examples_to_temp_folder():
+    #assert os.path.exists(EXAMPLES_path)
+    proc = None
+    base_projects_path = os.path.join(PATH, "pydss_projects")
+    print(f"Temporary path: {base_projects_path}")
+    os.makedirs(base_projects_path, exist_ok=True)
+    assert os.path.exists(base_projects_path)
+    copy_tree(EXAMPLES_path, base_projects_path)
+    return base_projects_path
+
+def test_external_interfaces_example():
+    example_name = "external_interfaces/pydss_project"
+    scenarios = [
+        {
+            'TOML': 'helics.toml',
+            'file': r"external_interfaces\Helics_example\run_dummy_federate.py",
+        },
+        {
+            'TOML': 'helics_itr.toml',
+            'file': r"external_interfaces\Helics_example\run_dummy_federate.py",
+        },
+        {
+            'TOML': 'simulation.toml',
+            'file': r"external_interfaces\Socket_example\run_socket_controller.py",
+        },
+        {
+            'TOML': None,
+            'file': r"external_interfaces\Python_example\run_pyDSS.py",
+        },
+    ]
+    run_example(example_name, scenarios)
+    return
+
+def test_monte_carlo_example():
+    example_name = "monte_carlo"
+    scenarios = [
+        {
+            'TOML': 'simulation.toml',
+            'file': None,
+        },
+    ]
+    run_example(example_name, scenarios)
+    return
+
+def test_dynamic_visualization_example():
+    example_name = "dynamic_visualization"
+    scenarios = [
+        {
+            'TOML' : 'simulation.toml',
+            'file' : None,
+        },
+        {
+            'TOML' : 'networkgraph.toml',
+            'file' : None,
+        },
+    ]
+    run_example(example_name, scenarios)
+    return
+
+def test_custom_contols_example():
+    example_name = "custom_contols"
+    scenarios = [
+        {
+            'TOML' : 'simulation.toml',
+            'file' : None,
+        },
+    ]
+    run_example(example_name, scenarios)
+    return
+
+def test_harmonics_example():
+    example_name = "harmonics"
+    scenarios = [
+        {
+            'TOML': 'simulation.toml',
+            'file': None,
+        },
+        {
+            'TOML': 'snapshot.toml',
+            'file': None,
+        },
+    ]
+    run_example(example_name, scenarios)
+    return
+
+def run_example(example_name, scenarios):
```

Comment:
Are there any files that need to be deleted after this runs?  I don't see that the fixture you created at the top of the file is being used.

Reply:
issue fixed